src/client/src/App/Trades/TradesHeader/ExcelButton.tsx

```diff
@@ -45,70 +43,48 @@ const Button = styled("button")`
     transform: scale(0.7);
   }
 `
-/**
- * Format trades data for CSV file, including escaping
- * inner commas.
- */
-export const getExportTable$ = (
-  trades$: Observable<Trade[]>,
-  colDef: ColDef,
-  colFields: (string | number)[],
-) =>
-  trades$.pipe(
-    map((trades) =>
-      trades.map((trade) =>
-        colFields.map((field) => {
-          let res =
-            colDef[field].excelValueFormatter?.(trade[field]) ??
-            colDef[field].valueFormatter?.(trade[field]) ??
-            trade[field]
-          if (typeof res === "string" && res?.includes(",")) {
-            res = '"' + res + '"'
-          }
-          return res
-        }),
-      ),
-    ),
-    take(1),
-  )
 
 const downloadCsv = (
-  trades$: Observable<Trade[]>,
+  trades: Trade[],
   colDef: ColDef,
   colFields: (string | number)[],
 ) => {
-  getExportTable$(trades$, colDef, colFields).subscribe((trades) => {
-    let csv = ""
-
-    // CSV header
-    colFields.forEach((field) => {
-      csv += colDef[field].headerName + ","
+  let csv = ""
```

Comment:
it's at times like this that I miss ag-grid (or similar) .. but I don't miss it most of the time 😆